### ISSUE TITLE
Virtual Terminal Sequences for Simple Cursor Positioning don't work

### ISSUE THREAD

**Issue by Toxaris** [microsoft/terminal#11005]
### Windows Terminal version (or Windows build number)

10.0.19042.1165

### Steps to reproduce

Activate virtual terminal processing and print `"Hello, world!\x1bD?"` to stdout.

### Expected Behavior

Based on the documentation about [virtual terminal sequences for simple cursor positioning](https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences#simple-cursor-positioning) I expect to see:

```
Hello, World?
```

That is, I expect that `\x1bD` moves the cursor left over the `!` and then the `?` is printed over the `!`.

In other words, I expect that `\x1bD` works like `\x1b[1D`.

### Actual Behavior

I see:

```
Hello, World!
             ?
```

**Comment by orcmid** [comment]
You need to contend with the fact that in C# the D is viewed as a hex digit and it is taken as part of the \x1bD.  I suggest you try a simple \x08 to get the backspace you want in this particular case.  Or use the official VT escape sequence.

**Comment by vefatica** [comment]
Or tell it where the escape sequence ends:  `Printf(L"Hello, world!\x1b\D?");`

**Comment by vefatica** [comment]
Or use a less troublesome version.

```
Printf(L"Hello, world!\x1b[D?");
Printf(L"Hello, world!\x1b[1D?");
```

**Comment by Toxaris** [comment]
Interesting point about the `\x` consuming the `D`. However, I'm using Rust, and I'm pretty sure that `\x` in Rust always consumes exactly the next two hexadecimal digits. See [Rust reference on character escapes](https://doc.rust-lang.org/stable/reference/tokens.html#character-escapes).

To double check, I tried the following program:

```rust
println!("Hello, world!\x1b[D?");

println!("Hello, world!\x1bD?");

print!("Hello, world!\x1b");
print!("D");
println!("?");
```

I see the following output in the integrated terminal in VS Code:

```
Hello, world?
             Hello, world?
                          Hello, world?
```

And the following output in Windows terminal:

```
Hello, world?
             Hello, world!
                          ?
                           Hello, world!
                                        ?
```

So in the integrated terminal in VS Code, `\x1bD` behaves like `\x1b[D`, but in Windows terminal, they behave differently.

**Comment by Toxaris** [comment]
I hunted around in the source code a bit and I found the following leads in `terminal/parser/OutputStateMachineEngine`:

- `EscActionCodes::IND_Index` is `D`:
https://github.com/microsoft/terminal/blob/0c901edd813f2bb61a52c3aca5384eb4d748b868/src/terminal/parser/OutputStateMachineEngine.hpp#L77

- `EscActionCodes::IND_Index` is dispatched as line feed without return (as expected from the name):
https://github.com/microsoft/terminal/blob/0c901edd813f2bb61a52c3aca5384eb4d748b868/src/terminal/parser/OutputStateMachineEngine.cpp#L220-L223)

It appears that the simple cursor positioning codes from the [article on docs.microsoft.com](https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences#simple-cursor-positioning) are simply not implemented in Windows terminal? Maybe because Windows terminal chooses to emulate a different terminal than described in that article?

**Comment by Toxaris** [comment]
> I suggest you try a simple \x08 to get the backspace you want in this particular case. Or use the official VT escape sequence.

Why is `ESC D` not an "official VT escape sequence"? It is one of the first sequences explained in the [documentation about virtual terminal sequences](https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences) which is prominently linked from [the recommendation to discontinue use of the classic console API](https://docs.microsoft.com/en-us/windows/console/classic-vs-vt).

**Comment by j4james** [comment]
Sorry for the confusion, but that's a mistake in the documentation. The first four "Simple Cursor Positioning" commands are actually VT52 sequences that only active in VT52 mode. In the default "ANSI" mode, they mean something completely different, although the only one that's currently implemented is `ESC D` (this is the "Index" command, which moves the cursor down).

**Comment by Toxaris** [comment]
Thanks @j4james, that makes sense. So the proper fix is to update the documentation. To my delight, I see that it is also on github, so I opened a PR. Interesting to see what will come of it.

**Comment by Toxaris** [comment]
Hmm wait this doesn't explain why the terminal integrated into VS Code interpretes the codes differently than Windows Terminal? Do they use different default configurations? Why?

**Comment by j4james** [comment]
> Do they use different default configurations? Why?

The behavior described in the documentation is the way the code used to work. That was corrected a while back, but those corrections take a while to make their way into the conhost that is included with Windows itself. Windows Terminal bundles its own build of conhost, so it always has the latest fixes. I'm guessing VS Code just uses the conhost included with Windows, or at least is not using the most recent version.

**Comment by DHowett** [comment]
Yep- VS Code uses the version of conhost that ships with Windows. James, I found that you noted this change in [the Compatibility section](https://github.com/microsoft/terminal/blob/main/doc/specs/%23976%20-%20VT52%20escape%20sequences.md#compatibility) of your spec document when you originally gave us VT52 support. I'm eternally grateful that we've written so much stuff down. 

@Toxaris sorry about the documentation miss, and thank you for bringing it to our attention 😄 

I'll close this one out when we merge the updated docs!
